docs/source/app-dev/bindings-scala/index.rst

```diff
@@ -0,0 +1,144 @@
+
+We recommend separating generated code and application code into different modules. There are two modules in the example below:
+
+- ``scala-codegen``
+    This modules contains all generated Scala classes.
+- ``application``
+    This is the application code that makes use of the generated Scala classes.
+
+.. literalinclude:: ./code-snippets/quickstart-scala/build.sbt
+   :start-after: // <doc-ref:modules>
+   :end-before: // </doc-ref:modules>
+
+``scala-codegen`` module uses the following function to generate Scala classes from a DAR file.
+
+.. literalinclude:: ./code-snippets/quickstart-scala/build.sbt
+   :start-after: // <doc-ref:generate-scala>
+   :end-before: // </doc-ref:generate-scala>
+
+Here is the entire `build.sbt file <https://github.com/digital-asset/daml/blob/master/language-support/scala/examples/quickstart-scala/build.sbt>`_
+
+Generating Scala Code from Command Line
+=======================================
+
+The above example demonstrates how to use Scala codegen from **sbt**. You can also call Scala codegen directly
+from a command line. For information on how to build a **DAR** file from a **DAML** model see :ref:`assistant-manual-building-dars`.
+
+You can download the `latest version <https://bintray.com/api/v1/content/digitalassetsdk/DigitalAssetSDK/com/daml/codegen-main/$latest/codegen-main-$latest.jar?bt_package=sdk-components>`_
+of the Scala codegen command line interface. Make sure that this versions is aligned with the version of the SDK you used to build the **DAR** file.
+
+To run Scala codegen::
+
+    $ java -jar <parth-to-codegen-main-jar> scala <path-to-DAR-file>=<package-name> \
+        --output-directory=<path-to-output-directory> --verbosity=<0|1|2|3|4>
+
+Here is an example, assuming SDK Version: **0.12.17**, DAR file: **./quickstart-scala.dar**,
+package name: **com.digitalasset.quickstart.iou.model**, codegen output directory: **./codegen-out** and
+verbosity level: **2** (INFO)::
+
+    $ java -jar codegen-main-100.12.17.jar scala ./quickstart-scala.dar=com.digitalasset.quickstart.iou.model \
+        --output-directory=./codegen-out --verbosity=2
+    ...
+    [INFO ] Scala Codegen result:
+    Number of generated templates: 3
+    Number of not generated templates: 0
+    Details:
+
+The output above tells that codegen produced Scala classes for 3 templates without errors (empty ``Details:`` line).
+
+Example code
+============
+
+This section refers to the IOU DAML example from the :doc:`Quickstart guide </getting-started/quickstart>` and
+`quickstart-scala sbt example <https://github.com/digital-asset/daml/tree/master/language-support/scala/examples/quickstart-scala>`_.
+
+Please keep in mind that all Scala examples compiled with ``-Xsource:2.13`` **scalac** option.
+
+Create a Contract and Send a Create Command
+-------------------------------------------
+
+To create a Scala class representing an ``Iou`` contract:
+::
+
+    import com.digitalasset.ledger.client.binding.{Primitive => P}
```

Comment:
Could you `literalinclude` this from somewhere rather than having it inline? Otherwise there's no guarantee this code won't break.